File: src/palindrome-linked-list-lcci.go
package main

import "fmt"

func main() {
	fmt.Println("vim-go")
}

/**
 * Definition for singly-linked list.
 */
type ListNode struct {
	Val  int
	Next *ListNode
}

// https://leetcode-cn.com/problems/palindrome-linked-list-lcci/
func isPalindrome(head *ListNode) bool {
	var stack []int
	for {
		if head == nil {
			break
		}
		if len(stack) == 0 {
			stack = append(stack, head.Val)
			head = head.Next
			continue
		}
		if head.Val == stack[len(stack)-1] {
			stack = stack[:len(stack)-1]
		} else {
			stack = append(stack, head.Val)
		}
		head = head.Next
	}
	return len(stack) == 0
}

File: src/palindrome-linked-list-lcci.py
#https://leetcode-cn.com/problems/palindrome-linked-list-lcci/
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None
class Solution:
    def isPalindrome(self, head: ListNode) -> bool:
        nums = []
        while head != None:
            nums.append(head.val)
            head = head.next
        for i in range(len(nums)):
            if nums[i] != nums[-1-i]:
                return False
            if i >= len(nums)-i-1:
                break
        return True

